### PR TITLE
Abuse-c validation comment must refer to the LIR organisation

### PR DESCRIPTION
**Pull request by eshryane**
Abuse-c validation comment must refer to the LIR organisation, if the address is suspect.

### CODE REVIEW

**Comment by eshryane** [comment]
Waiting for integration tests to pass

File: whois-query/src/main/java/net/ripe/db/whois/query/dao/jdbc/JdbcAbuseValidationStatusDao.java [hunk 1]
```diff
@@ -23,10 +24,10 @@ public JdbcAbuseValidationStatusDao(@Qualifier("internalsDataSource") final Data
     }
 
     @Override
-    public boolean isSuspect(CIString email) {
-        return internalsTemplate.queryForObject(
-                "select count(*) from abuse_email where address = ? and status = ?",
-                new String[] { email.toLowerCase(), "SUSPECT" },
+    public boolean isSuspect(@Nullable final CIString address) {
+        return (address != null) && internalsTemplate.queryForObject(
```
Comment (sbusk) [inline whois-query/src/main/java/net/ripe/db/whois/query/dao/jdbc/JdbcAbuseValidationStatusDao.java]: why the null check?

File: whois-update/src/main/java/net/ripe/db/whois/update/handler/validator/organisation/LirMntByAttributeCountValidator.java [hunk 2]
```diff
@@ -37,7 +36,7 @@ public LirMntByAttributeCountValidator(final Maintainers maintainers) {
     @Override
     public void validate(final PreparedUpdate update, final UpdateContext updateContext) {
         final RpslObject originalObject = update.getReferenceObject();
-        if (!LIR.equals(originalObject.getValueForAttribute(AttributeType.ORG_TYPE))) {
+        if (! OrgType.LIR.getName().equals(originalObject.getValueForAttribute(AttributeType.ORG_TYPE))) {
```
Comment (sbusk) [inline whois-update/src/main/java/net/ripe/db/whois/update/handler/validator/organisation/LirMntByAttributeCountValidator.java]: LIR == OrgType.getFor(originalObject.getValueForAttribute(AttributeType.ORG_TYPE)) is a bit cleaner..?

File: whois-query/src/main/java/net/ripe/db/whois/query/dao/jdbc/JdbcAbuseValidationStatusDao.java (hunk 1 above)
Comment (eshryane) [reply whois-query/src/main/java/net/ripe/db/whois/query/dao/jdbc/JdbcAbuseValidationStatusDao.java]: I added more input checks, but it's not necessary in this case, I removed it again

File: whois-update/src/main/java/net/ripe/db/whois/update/handler/validator/organisation/LirMntByAttributeCountValidator.java (hunk 2 above)
Comment (eshryane) [reply whois-update/src/main/java/net/ripe/db/whois/update/handler/validator/organisation/LirMntByAttributeCountValidator.java]: Agreed, refactored in all validators to be consistent.

**Comment by eshryane** [comment]
Integration tests have all passed

File: whois-query/src/main/java/net/ripe/db/whois/query/planner/AbuseCFinder.java [hunk 3]
```diff
@@ -46,61 +47,97 @@ public AbuseCFinder(final RpslObjectDao objectDao,
...
         }
 
-        final RpslObject responsibleOrg = findResponsibleOrgObject(object);
-        return responsibleOrg != null? responsibleOrg.getValueForAttribute(AttributeType.ORG) : null;
+        return findResponsibleOrgReference(rpslObject);
     }
 
     @Nullable
-    private RpslObject findResponsibleOrgObject(final RpslObject rpslObject) {
+    private CIString findResponsibleOrgReference(final RpslObject rpslObject) {
```
Comment (sbusk) [inline whois-query/src/main/java/net/ripe/db/whois/query/planner/AbuseCFinder.java]: findResponsibleOrgReference() and findResponsibleLirOrgReference() are practically identical apart from findResponsibleLirOrgReference() checking the org is an lir, maybe have just one method which also takes on the org type we're looking for as an argument so we don't duplicate the logic..

Comment (eshryane) [reply whois-query/src/main/java/net/ripe/db/whois/query/planner/AbuseCFinder.java]: Agreed, I refactored the methods

**Comment by eshryane** [comment]
Integration tests have all passed

**Pull request merged**